src/OSPSuite.Utility/FileHelper.cs

```diff
@@ -222,11 +222,45 @@ public static bool HasWriteAccessToFolder(string folderPath)
       }
 
       /// <summary>
-      ///    Returns true if the path is read only or do not have access to view the permissions otherwise false
+      ///    Returns <c>false</c> if the path is read only or do not have access to view the permissions otherwise <c>true</c>
       /// </summary>
-      public static bool HasWriteAccessToFolder(DirectoryInfo directoryInfo)
+      public static bool HasWriteAccessToFolder(DirectoryInfo directoryInfo) =>
+         HasWriteAccessToFolder(directoryInfo.FullName);
+
+      /// <summary>
+      ///    Copies the <paramref name="source" /> directory into the <paramref name="target" /> directory. If
+      ///    <paramref name="createRootDirectory" /> is <c>true</c> (default), a sub folder named after the source
+      ///    folder will be created. Otherwise the content of the  <paramref name="source" /> directory will be directly copied
+      ///    under  the <paramref name="target" /> directory.
+      /// </summary>
+      public static void CopyDirectory(string source, string target, bool createRootDirectory = true) =>
+         CopyDirectory(new DirectoryInfo(source), new DirectoryInfo(target), createRootDirectory);
+
+      /// <summary>
+      ///    Copies the <paramref name="source" /> directory into the <paramref name="target" /> directory. If
+      ///    <paramref name="createRootDirectory" /> is <c>true</c> (default), a sub folder named after the source
+      ///    folder will be created. Otherwise the content of the  <paramref name="source" /> directory will be directly copied
+      ///    under  the <paramref name="target" /> directory.
+      /// </summary>
+      public static void CopyDirectory(DirectoryInfo source, DirectoryInfo target, bool createRootDirectory = true)
       {
-         return HasWriteAccessToFolder(directoryInfo.FullName);
+         if (!target.Exists)
+            target.Create();
+
+         var rootTarget = target;
+         if (createRootDirectory)
+            rootTarget = target.CreateSubdirectory(source.Name);
+
+         foreach (var dir in source.GetDirectories())
+         {
+            //Do not creaet root directory as it is created in this call
+            CopyDirectory(dir, rootTarget.CreateSubdirectory(dir.Name), createRootDirectory: false);
```

Comment:
why not just 
`CopyDirectory(dir, rootTarget)` instead of 
`CopyDirectory(dir, rootTarget.CreateSubdirectory(dir.Name), createRootDirectory: false);` ?
Result would be the same.